### DISCUSSION TITLE
PHP warning when using parseContentShortcut(): Passing null to parameter

### DISCUSSION THREAD

**Discussion by keycolour** [datenstrom/community#884] [Report a bug] [answered]
I'm developing a blog section for my website, and wanted to display my blog tags as a list on the blog-start layout page. I used the code `<?php echo $this->yellow->page->parseContentShortcut("blogtags", null, "inline"); ?>` and it worked! (It displays as a bulleted list, but that can be changed.)

However, doing this also displayed the warning, "Deprecated: trim(): Passing null to parameter #1 ($string) of type string is deprecated in [path]\system\extensions\core.php on line 2335" (full path altered for privacy).

So I investigated line 2335 and with a bit of Google-fu found that changing `trim($text));` on that line to `trim($text ?? ''));` makes the warning go away. Changing it to `trim((string) $text));` also works, although I'm not a PHP coder so I wouldn't know which of the two would be better.

Anyway, just wanted to pass that along!

**Comment by markseuffert** [comment, marked as answer]
I checked the source code and think this is not a bug in the core. I suspect the calling code is wrong. For newer PHP versions `null` and strings are non-interchangeable types, there's no automatic type cast anymore, so I think the second argument in the calling code must be a string. Sometimes it is enough to manually replace `null` by `""`.

Can you use `parseContentShortcut("blogtags", "", "inline")` and check if this fixes the problem?

**Reply by markseuffert** [reply]
Can someone else be so kind and check if this fixes the problem?

**Comment by schulle4u** [comment]
During my extensive tests for datenstrom/community#800 lately I also came across this message and can confirm that passing either shortcut parameters or just an empty string to the calling code will fix the warning.
